### PR TITLE
fix(version): implement `helm version -c`, mark as hidden

### LINKED ISSUE #6322: Feature request: a `helm version` option that works consistently across v2 and v3
I am consuming Helm programmatically and sometimes need to issue slightly different command syntax according to whether the user has Helm 2 or Helm 3.  I can try to detect this using `helm version` _but_...

* On Helm 2, `helm version` attempts to reach Tiller, which is potentially slow.  I can avoid this with `helm version -c` but...
* On Helm 3, `helm version -c` is an error.
* On Helm 2, `helm version --short` returns a string of the form `Client: v2.14.3+g0e7f3b6` (plus server info which can be suppress with `-c`)
* On Helm 3, `helm version --short` returns a string of the form `v3.0.0-beta.2+g26c7338` (no `Client: ` prefix)

All this makes detecting v2 vs v3 a bit fiddly.  I could issue `helm version -c` and assume v3 if I get an error, but perhaps it would be nice to continue to support `-c` as a hidden flag and have it force the output to the same format as v2?

### PR DESCRIPTION
**Pull request by bacongobbler**
Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

supersedes #6368
closes #6322

### CODE REVIEW

File: cmd/helm/version.go [hunk 1]
```diff
@@ -59,8 +59,10 @@ func newVersionCmd(out io.Writer) *cobra.Command {
 		},
 	}
 	f := cmd.Flags()
+	f.BoolP("client-only", "c", true, "display client version information")
```
Comment (marckhouzam) [inline cmd/helm/version.go]: Isn't the flag in helm v2 named `client`?

Comment (marckhouzam) [reply cmd/helm/version.go]: `--client-only` did feel familiar but I couldn't quite figure out why.
Then I searched the code and found that it is `helm init --client-only` but `helm version --client` 😉

File: cmd/helm/version.go [hunk 2]
```diff
@@ -59,8 +59,10 @@ func newVersionCmd(out io.Writer) *cobra.Command {
...
 	f.BoolVar(&o.short, "short", false, "print the version number")
 	f.StringVar(&o.template, "template", "", "template for version string format")
+	f.MarkHidden("client-only")
```
Comment (marckhouzam) [inline cmd/helm/version.go]: 👍 for hidden

File: cmd/helm/version.go (hunk 1 above)
Comment (bacongobbler) [reply cmd/helm/version.go]: Gah. I'm mixing flags again -_-

Thanks for catching this.

**Review by hickeyma — DISMISSED** [review]
@bacongobbler Initially I was ok with adding this capability in but during the review I am thinking this is not such a good idea.

My reasoning is as follows:
We are adding a flag which is artificial and contrived in Helm 3 as there is no client and server anymore. There is just helm.  Helm 3 removes Tiller and the client/server architecture and therefore all UX around this. When we add this `--c` flag, we are reintroducing a workaround for a niche case that some users used in v2.  I know that `helm version` has roundtrip issues with Tiller in v2 but it provides the capability in a standard UX fashion.
I am therefore not sure that this justifies putting this in place.

**Comment by thomastaylor312** [comment]
I think I agree with @hickeyma on this one. I feel like adding a no-op flag for something now non-existent like Tiller that is essentially The Thing™ people have been wanting Helm 3 is probably not the best. This would be helping one specific use case that can still be easily worked around, albeit in a slightly verbose/clunky fashion

**Comment by bacongobbler** [comment]
Okay. Closing as there is consensus to not go ahead with this feature.

**Pull request closed without merging**